src/mission.cpp

```diff
@@ -389,14 +390,21 @@ bool mission::is_complete( const character_id &_npc_id ) const
             }
             const inventory &tmp_inv = player_character.crafting_inventory();
             // TODO: check for count_by_charges and use appropriate player::has_* function
+            search_modifier::mode = 0b100001;
             if( !tmp_inv.has_amount( type->item_id, item_count ) ) {
-                return tmp_inv.has_amount( type->item_id, 1 ) && tmp_inv.has_charges( type->item_id, item_count );
+                ret = tmp_inv.has_amount( type->item_id, 1 ) && tmp_inv.has_charges( type->item_id, item_count );
+                search_modifier::mode = 0b1;
+                return ret;
             }
+            search_modifier::mode = 0b1;
         }
         return true;
 
         case MGOAL_FIND_ANY_ITEM:
-            return player_character.has_mission_item( uid ) && ( !npc_id.is_valid() || npc_id == _npc_id );
+            search_modifier::mode = 0b100001;
```

Comment:
I would not recommend using a magic number and global variable for the search mode.